### DISCUSSION TITLE
Pocketbase for NextJs SSG: opinion?

### DISCUSSION THREAD

**Discussion by osseonews** [pocketbase/pocketbase#597] [Q&A]
For the people who are much more knowledgeable than me, isn't SQLite with Pocketbase ideal for static site generation (SSG) on platforms like NextJs? In this case, all the pages are built on the server, during the build phase, so there would be no concurrency risk whatsoever as users view your site, as every page is already pre-built and cached. I can understand why people are concerned about using something like SQLite for dynamic parts of a page, but what is argument against it for pure static site generation? The database is pretty much only hit on the server during the build phase.  And even for updating with NextJs you can just use their new on demand revalidation to revalidate specific pages when the Pocketbase database is updated, again reducing any concurrency risk. Is the above correct?

**Comment by ganigeorgiev** [comment]
>  I can understand why people are concerned about using something like SQLite for dynamic parts of a page

Those concerns are usually not backed by any real tests. SQLite supports multiple readers and a single writer (multiple writers just queue up) and **in WAL mode the readers and writer don't block each others** - https://sqlite.org/wal.html.

In fact, for read heavy apps you'll have better performance with SQLite compared to the other traditional databases.
[Many small queries are efficient in SQLite](https://www.sqlite.org/np1queryprob.html):
> 200 SQL statements per webpage is excessive for client/server database engines like MySQL, PostgreSQL, or SQL Server.
> **But with SQLite, 200 or more SQL statement per webpage is not a problem.**

---

I'm not sure what are your requirements, but for me SQLite is NOT suitable when:
- you have an extremely write heavy app (eg. keeping logs; PocketBase uses a separate logs database file with lazy writes to avoid blocking the primary db operations)
- you have long running transactions that span more than a few seconds (in PocketBase those are kept to minimum)
- you need horizontal scaling and row-based logging for replication out of the box (for SQLite there are 3rd party solutions)

You could also read the official SQLite "When to use" guide - https://www.sqlite.org/whentouse.html
